nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchHttp.java

```diff
@@ -321,9 +321,16 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             final Response getResponse;
             try {
                 getResponse = sendRequestToElasticsearch(okHttpClient, url, username, password, "PUT", requestBody);
-            } catch (IllegalStateException | IOException ioe) {
-                throw new ProcessException(ioe);
+            } catch (final Exception e) {
+                logger.error("Routing to {} due to exception: {}", new Object[]{REL_FAILURE.getName(), e}, e);
+                flowFilesToTransfer.forEach((flowFileToTransfer) -> {
+                    flowFileToTransfer = session.penalize(flowFileToTransfer);
```

Comment:
I think it was because the processor couldn't tell from the exception whether it could/should be handled as a retry or failure, so I just penalized it and sent it to failure. I can change it if that's not the best way to handle it